include/ginkgo/core/solver/cg.hpp

```diff
@@ -116,6 +116,22 @@ class Cg : public EnableLinOp<Cg<ValueType>>,
     GKO_ENABLE_LIN_OP_FACTORY(Cg, parameters, Factory);
     GKO_ENABLE_BUILD_METHOD(Factory);
 
+    constexpr static int num_work_vectors = 10;
+    constexpr static int vector_residual = 0;
+    constexpr static int vector_preconditioned_residual = 1;
+    constexpr static int vector_search_direction = 2;
+    constexpr static int vector_search_direction2 = 3;
+    constexpr static int scalar_alpha = 4;
+    constexpr static int scalar_beta = 5;
+    constexpr static int scalar_prev_rho = 6;
+    constexpr static int scalar_rho = 7;
+    constexpr static int scalar_one = 8;
+    constexpr static int scalar_minus_one = 9;
```

Comment:
Nice. I think we can even use this to have some automatic memory transfer, BW and Flop estimates.

Reply:
Let me just point to #802 for that :)